### PR TITLE
fix particle-system deserialization

### PR DESCRIPTION
**Pull request by SantyWang**
Re: https://github.com/cocos/cocos-engine/issues/12047

### Changelog

* fix the issue that enum and bitmask data deserialization incorrect

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/core/data/deserialize-dynamic.ts [hunk 1]
```diff
@@ -145,11 +145,11 @@ function compileDeserializeJIT (self: _Deserializer, klass: CCClassConstructor<u
 
         // function undefined object(null) string boolean number
         const defaultValue = CCClass.getDefault(attrs[propName + POSTFIX_DEFAULT]);
-        const userType = attrs[propName + POSTFIX_TYPE] as AnyFunction | undefined;
+        const userType = attrs[propName + POSTFIX_TYPE] as AnyFunction | string | undefined;
         if (fastMode && (defaultValue !== undefined || userType)) {
             let isPrimitiveType;
             if (defaultValue === undefined) {
-                isPrimitiveType = userType instanceof Attr.PrimitiveType;
```
Comment (jareguo) [inline cocos/core/data/deserialize-dynamic.ts:152]: 为什么不同样继承自 Attr.PrimitiveType？

Comment (SantyWang) [reply cocos/core/data/deserialize-dynamic.ts:152]: enum 要怎么继承这个，而且引擎这么多 enum 呢

**Review by shrinktofit — APPROVED** [review]

File: cocos/core/data/deserialize-dynamic.ts (hunk 1 above)
Comment (jareguo) [reply cocos/core/data/deserialize-dynamic.ts:152]: 其实不叫继承，只是 new 一个 object，通过 instanceof 来得出是不是 primitive type。
`export const CCEnum = new PrimitiveType('Enum', 0);`
然后这个 CCEnum 应该就能取代 EnumTag 的功能。
但是我看现在的代码好像有些矛盾的地方……

算了还是不改了吧

Comment (SantyWang) [reply cocos/core/data/deserialize-dynamic.ts:152]: 企业微信截图_20d21d15-893b-49ae-ac29-dbe1d8ab59d8
枚举没办法指定这个默认值，而且单一默认值对枚举没有意义

Comment (jareguo) [reply cocos/core/data/deserialize-dynamic.ts:152]: 默认值0就行了，就像ccinteger和ccstring一样

Comment (SantyWang) [reply cocos/core/data/deserialize-dynamic.ts:152]: 默认值为 0 不行吧，引擎里面有取默认值的地方，如果把 0 赋值给了一个 string 类型的枚举，会出现什么样的异常无法确定啊

**Review by jareguo — APPROVED** [review]

**Pull request merged**